File: src/main/java/br/unifor/costify/application/contracts/IngredientRepository.java
package br.unifor.costify.application.contracts;

import br.unifor.costify.domain.entity.Ingredient;
import br.unifor.costify.domain.valueobject.Id;
import java.util.Optional;

public interface IngredientRepository {

  Ingredient save(Ingredient ingredient);

  Optional<Ingredient> findById(Id id);

  boolean existsByName(String name);

  void deleteById(Id id);
}

File: internal/application/contract/ingredient_repository.go
package contract

import (
	"github.com/vini/costify-go/internal/domain/entity"
	"github.com/vini/costify-go/internal/domain/valueobject"
)

// IngredientRepository defines the contract for ingredient persistence operations
type IngredientRepository interface {
	Save(ingredient *entity.Ingredient) (*entity.Ingredient, error)
	FindById(id valueobject.Id) (*entity.Ingredient, error)
	ExistsByName(name string) (bool, error)
	DeleteById(id valueobject.Id) error
}
